plugins/modules/s3_object.py

```diff
@@ -626,7 +641,10 @@ def create_dirkey(module, s3, bucket, obj, encrypt, expiry):
 
     url = put_download_url(module, s3, bucket, obj, expiry)
```

Comment:
This call duplicates the try/except block above it.  The try/except block above (lines 635 through 640) should probably be removed